server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -0,0 +1,170 @@
+
+    @Inject
+    @VisibleForTesting
+    SetMailboxesUpdateProcessor(MailboxUtils<Id> mailboxUtils, MailboxManager mailboxManager) {
+        this.mailboxUtils = mailboxUtils;
+        this.mailboxManager = mailboxManager;
+    }
+
+    @Override
+    public SetMailboxesResponse process(SetMailboxesRequest request, MailboxSession mailboxSession) {
+        SetMailboxesResponse.Builder responseBuilder = SetMailboxesResponse.builder();
+        request.getUpdate().entrySet().stream().forEach(update -> handleUpdate(update, responseBuilder, mailboxSession));
+        return responseBuilder.build();
+    }
+
+    private void handleUpdate(Entry<String, MailboxUpdateRequest> update, Builder responseBuilder, MailboxSession mailboxSession) {
+        String mailboxId = update.getKey();
+        MailboxUpdateRequest updateRequest = update.getValue();
+        try {
+            validateMailboxName(updateRequest, mailboxSession);
+            Mailbox mailbox = getMailbox(mailboxId, mailboxSession);
+            validateParent(mailbox, updateRequest, mailboxSession);
+
+            updateMailbox(mailbox, updateRequest, mailboxSession);
+            responseBuilder.updated(update.getKey());
+        } catch (MailboxNameException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("invalidArguments")
+                    .description(e.getMessage())
+                    .build());
+        } catch (MailboxNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The mailbox '%s' was not found", mailboxId))
+                    .build());
+        } catch (MailboxParentNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The parent mailbox '%s' was not found.", e.getParentId()))
+                    .build());
+        } catch (MailboxHasChildException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("invalidArguments")
+                    .description("Cannot update a parent mailbox.")
+                    .build());
+        } catch (MailboxException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("anErrorOccurred")
+                    .description("An error occurred when updating the mailbox")
+                    .build());
+        }
+    }
+
+    private Mailbox getMailbox(String mailboxId, MailboxSession mailboxSession) throws MailboxNotFoundException {
+        return mailboxUtils.mailboxFromMailboxId(mailboxId, mailboxSession)
+                .orElseThrow(() -> new MailboxNotFoundException(mailboxId));
+    }
+
+    private void validateMailboxName(MailboxUpdateRequest updateRequest, MailboxSession mailboxSession) throws MailboxNameException {
```

Comment:
We should document the limitation in the specification; then we should handle (or contribute) in another issue.